lib/astarte_data_access.ex

```diff
@@ -0,0 +1,44 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2023 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.DataAccess do
+  # Automatically defines child_spec/1
+  use Supervisor
+  alias Astarte.DataAccess.Config
+
+  def start_link(init_arg) do
+    Supervisor.start_link(__MODULE__, init_arg, name: __MODULE__)
+  end
+
+  @impl true
+  def init(init_arg) do
+    atom_keys? = Keyword.get(init_arg, :atom_keys, false)
+
+    xandra_options =
+      Config.xandra_options!()
+      |> Keyword.put(:name, :astarte_data_access_xandra)
+      |> Keyword.put(:atom_keys, atom_keys?)
```

Comment:
This should be passed from the outside: it's strange that _a library_ reads the system env to configure itself, that should be done by the application.

So here there should be something like
```elixir
xandra_options =
  Keyword.get(init_arg, :xandra_opts, [])
  |> Keyword.put(:name, :astarte_data_access_xandra)
  |> Keyword.put(:atom_keys, atom_keys?)
```

and then in the application using the library
```elixir

DataAccessConfig.validate!()
...
xandra_opts = Config.xandra_options!()
data_access_options = [xandra_opts: xandra_opts]
# If another Xandra connection is present in the app it would also do this
other_xandra_opts = Keyword.put(xandra_opts, :name, :other_xandra_conn)

children = [
  {Xandra.Cluster, other_xandra_opts},
  {Astarte.DataAccess, data_access_opts}
  ...
]
```

In general, configuration should always be passed top-down from the application downward as much as possible